src/cpp/server/server-options.json

```diff
@@ -113,6 +113,14 @@
             "skipAccessorGeneration": true,
             "description": "The port that RStudio Server will bind to while listening for incoming connections. If left empty, the port will be automatically determined based on your SSL settings (443 for SSL, 80 for no SSL)."
          },
+         {
+            "name": "www-socket",
+            "memberName": "wwwSocket_",
+            "type": "string",
+            "defaultValue": "",
+            "skipAccessorGeneration": true,
+            "description": "The socket that RStudio Server will bind to while listening for incoming connections. If left empty, a port will be used."
```

Comment:
Does this mean if (from a shell script) `rserver --www-socket ''` (or `rserver --www-socket`, without any option-argument) is specified, then rserver would listen on a TCP port instead? It might be preferable to throw an error in this case to catch any "accidents" (e.g., `rserver --www-socket $SOCKET_PATH` where SOCKET_PATH variable is inadvertently unset or an empty string)

Reply:
With `if (options().wwwSocket() != "")` in `ServerInit.cpp` a TCP port is choosen on `rserver --www-socket ''` or `rserver --www-socket`.